File: platforms/documentation/docs/src/samples/templates/gradle-plugin-in-java/src/functionalTest/java/com/example/plugin/GreetingPluginFunctionalTest.java
package com.example.plugin;

import org.gradle.testkit.runner.BuildResult;
import org.gradle.testkit.runner.GradleRunner;
import org.junit.Test;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.io.Writer;
import java.nio.file.Files;

import static org.junit.Assert.assertTrue;


public class GreetingPluginFunctionalTest {
    @Test
    public void canRunTask() throws IOException {
        // Setup the test build
        File projectDir = new File("build/functionalTest");
        Files.createDirectories(projectDir.toPath());
        writeString(new File(projectDir, "settings.gradle"), "");
        writeString(new File(projectDir, "build.gradle"),
            "plugins {" +
                "  id('com.example.plugin.greeting')" +
                "}");

        // Run the build
        BuildResult result = GradleRunner.create()
            .forwardOutput()
            .withPluginClasspath()
            .withArguments("greet")
            .withProjectDir(projectDir)
            .build();

        // Verify the result
        assertTrue(result.getOutput().contains("Hello from plugin 'com.example.plugin.greeting'"));
    }

    private void writeString(File file, String string) throws IOException {
        try (Writer writer = new FileWriter(file)) {
            writer.write(string);
        }
    }
}

File: platforms/documentation/docs-site/src/samples/userguide/reference/testing-gradle-plugins/plugin-binary/kotlin/plugin/src/functionalTest/kotlin/org/example/GreetingPluginFunctionalTest.kt
package org.example

import java.io.File
import kotlin.test.assertTrue
import org.gradle.testkit.runner.GradleRunner
import org.gradle.testkit.runner.TaskOutcome
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.io.TempDir

class GreetingPluginFunctionalTest {

    @field:TempDir
    lateinit var projectDir: File

    private val settingsFile by lazy { projectDir.resolve("settings.gradle.kts") }
    private val buildFile by lazy { projectDir.resolve("build.gradle.kts") }

    @Test
    fun `can run greeting task`() {
        // Arrange: write a tiny build that applies the plugin
        settingsFile.writeText("") // single-project build
        buildFile.writeText(
            """
            plugins {
                id("org.example.greeting")
            }
            """.trimIndent()
        )

        // Act: execute the task in an isolated Gradle build
        val result = GradleRunner.create()
            .withProjectDir(projectDir)
            .withPluginClasspath()      // picks up your plugin-under-test from the test classpath
            .withArguments("greet")
            .forwardOutput()
            .build()

        // Assert: verify console output and successful task outcome
        assertTrue(result.task(":greet")?.outcome == TaskOutcome.SUCCESS)
    }
}
